src/python/pants/backend/python/goals/export.py

```diff
@@ -230,30 +260,83 @@ async def do_export(
+                    ),
+                ),
+                {
+                    **complete_pex_env.environment_dict(python=requirements_pex.python),
+                    "PEX_MODULE": "pex.tools",
+                },
+            ),
+            # Remove the requirements and pex pexes, to avoid confusion.
+            PostProcessingCommand(["rm", "-rf", tmpdir_under_digest_root]),
+        ]
+
+        if req.editable_local_dists_digest is not None:
+            # insert editable wheel post processing commands
+            wheels_snapshot = await Get(Snapshot, Digest, req.editable_local_dists_digest)
+            py_minor_version = ".".join(py_version.split(".", 2)[:2])
+            lib_dir = os.path.join(output_path, "lib", f"python{py_minor_version}", "site-packages")
+            dist_info_dirs = [
+                os.path.join(lib_dir, "-".join(f.split("-")[:2]) + ".dist-info")
+                for f in wheels_snapshot.files
+            ]
+            post_processing_cmds[1:1] = [
+                PostProcessingCommand(
+                    [
+                        # the wheels are "sources" in the pex and get dumped in lib_dir
+                        # so we move them to tmpdir where they will be removed at the end.
+                        "mv",
+                        *(os.path.join(lib_dir, f) for f in wheels_snapshot.files),
+                        tmpdir_under_digest_root,
+                    ]
+                ),
+                PostProcessingCommand(
+                    [
+                        # now install the editable wheels
+                        os.path.join(output_path, "bin", "pip"),
+                        "install",
+                        "--no-deps",  # deps already installed via requirements.pex
+                        "--no-build-isolation",  # avoid vcs dep downloads (as they are installed)
+                        *(os.path.join(tmpdir_under_digest_root, f) for f in wheels_snapshot.files),
+                    ]
+                ),
+                PostProcessingCommand(
+                    [
+                        # replace pip's direct_url.json (which points to the wheel)
+                        # with ours (which points to build_dir sources and is marked "editable").
+                        "sh",
+                        "-c",
+                        " ".join(
+                            [
+                                f"mv -f {src} {dst};"
+                                for src, dst in zip(
+                                    [
+                                        os.path.join(d, "direct_url__pants__.json")
+                                        for d in dist_info_dirs
+                                    ],
+                                    [os.path.join(d, "direct_url.json") for d in dist_info_dirs],
+                                )
+                            ]
+                        ),
+                    ]
+                ),
+            ]
```

Comment:
Yeah. This was the simplest/clearest method I found so far. But it could use some more refactoring to feel even better. Next time someone has to mess with virtualenv exports, maybe they'll see a better way to do this.